apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/AccountControllerTest.java

```diff
@@ -681,6 +694,49 @@ public void check2FA_withoutRequestAttribute_thenGetError_1000(String uri) throw
         checkMandatoryParameterMissingErrorCode(response, 1000);
     }
 
+    @Test
+    void counts_SUCCESS() throws URISyntaxException, IOException {
+        String accountCountUri = "/accounts/count";
+
+        // prepare data
+        AccountsCountDto dto = new AccountsCountDto(112L, 113L, 1, 124L);
+        AccountEffectiveBalanceDto balanceDto = new AccountEffectiveBalanceDto(
+            100L, 200L, 100L, 200L, 200L, "123", "RS-ADVB");
+        dto.topHolders.add(balanceDto);
+        doReturn(dto).when(accountStatisticsService).getAccountsStatistic(Constants.MIN_TOP_ACCOUNTS_NUMBER);
+        // init mocks
+//        ServerInfoController controller = new ServerInfoController(serverInfoService);
+        dispatcher.getRegistry().addSingletonResource(endpoint);
+        // call
+        String uri = accountCountUri + "?numberOfAccounts=" + Constants.MIN_TOP_ACCOUNTS_NUMBER;
+        MockHttpRequest request = MockHttpRequest.get(uri);
+        MockHttpResponse response = new MockHttpResponse();
+        dispatcher.invoke(request, response);
+        // check
+        assertEquals(200, response.getStatus());
+        String respondJson = response.getContentAsString();
+        AccountsCountDto dtoResult = mapper.readValue(respondJson, new TypeReference<>(){});
+        assertNotNull(dtoResult.topHolders);
+        assertEquals(112L, dtoResult.totalSupply);
+        assertEquals(1, dtoResult.topHolders.size());
+
+        // call 2
+        uri = accountCountUri + "?numberOfAccounts="; // empty value
```

Comment:
Better to separate test cases or use @ParametrizedTest to supply other parameters causing same behavior

Reply:
updated to @ParametrizedTest